bsp/esp_bsp_devkit/src/esp_bsp_devkit.c

```diff
@@ -400,9 +405,36 @@ esp_err_t bsp_iot_button_create(button_handle_t btn_array[], int *btn_cnt, int b
         *btn_cnt = 0;
     }
     for (int i = 0; i < BSP_BUTTON_NUM; i++) {
-        btn_array[i] = iot_button_create(&bsp_button_config[i]);
-        if (btn_array[i] == NULL) {
-            ret = ESP_FAIL;
+        switch (i) {
```

Comment:
Can you simplify it? (remove `switch`) and make somethink like that:
```
if (CONFIG_BSP_BUTTONS_NUM > i) {
ret |= bsp_iot_button_create_internal(i+1);
}
```

Reply:
Also can't do it, Because Macro definitions need to be determined at compile time.